internal/compiler-bridge/src/main/scala/xsbt/ExtractAPI.scala

```diff
@@ -61,17 +64,18 @@ class ExtractAPI[GlobalType <: Global](
   // this cache reduces duplicate work both here and when persisting
   //   caches on other structures had minimal effect on time and cache size
   //   (tried: Definition, Modifier, Path, Id, String)
-  private[this] val typeCache = perRunCaches.newMap[(Symbol, Type), xsbti.api.Type]()
+
+  private[this] val typeCache = new java.util.HashMap[(Symbol, Type), xsbti.api.Type]()
   // these caches are necessary for correctness
-  private[this] val structureCache = perRunCaches.newMap[Symbol, xsbti.api.Structure]()
+  private[this] val structureCache = new java.util.HashMap[Symbol, xsbti.api.Structure]()
```

Comment:
The motivation for using Java maps was to avoid the cost of generalized equality and hashing. Usually I'd reach for an `AnyRefMap`, but AFAICT this code needs to cross compile to Scala 2.10, so it was easier to just use a Java Map.